Bindings/Java/OpenSimJNI/OpenSimContext.cpp

```diff
@@ -124,13 +125,21 @@ double OpenSimContext::getMuscleLength(Muscle& m) {
 }
 
 const Array<AbstractPathPoint*>& OpenSimContext::getCurrentPath(Muscle& m) {
-  return m.getGeometryPath().getCurrentPath(*_configState);
+  if (auto* p = dynamic_cast<const GeometryPath*>(&m.getPath())) {
+      return p->getCurrentPath(*_configState);
+  } else {
+      OPENSIM_THROW(Exception, "Muscle path is not a GeometryPath.")
+  }
 }
 
 void OpenSimContext::copyMuscle(Muscle& from, Muscle& to) {
-  to = from;
-  recreateSystemKeepStage();
-  to.updGeometryPath().updateGeometry(*_configState);
+  if (auto* p = dynamic_cast<const GeometryPath*>(&from.getPath())) {
```

Comment:
Similar to above, reverting these changes since I've reverted the `updGeometryPath` change. Will discuss with @aymanhab what makes sense for a refactor here.

Reply:
Will resolve this conversation as the method will  be retired or rewritten